src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -2,25 +2,47 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using System.Collections.Generic;
-using System.CommandLine;
-using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Diagnostics.Monitoring;
+using Microsoft.Diagnostics.Monitoring.Egress;
 using Microsoft.Diagnostics.Monitoring.RestServer;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Microsoft.Diagnostics.Tools.Monitor
 {
     internal sealed class DiagnosticsMonitorCommandHandler
     {
         private const string ConfigPrefix = "DotnetMonitor_";
-        private const string ConfigPath = "/etc/dotnet-monitor";
+        private const string SettingsFileName = "settings.json";
+        private const string ProductFolderName = "dotnet-monitor";
+
+        // Location where shared dotnet-monitor configuration is stored.
+        // Windows: "%ProgramData%\dotnet-monitor
+        // Other: /etc/dotnet-monitor
+        private static readonly string SharedConfigDirectoryPath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+            Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), ProductFolderName) :
```

Comment:
While /etc/dotnet-monitor makes a lot of sense for container scenarios, how does programdata\dotnet-monitor work on windows? For example, if one user creates a settings.json file in that location, does this preclude another user from changing the egress configuration?

Reply:
%ProgramData% is a restricted location (only admins can change it, but anyone should be able to read it). This is the same as /etc on Linux where only root (or a super user) can change files in this location. Both of these locations are machine-wide configuration data locations. It just so happens that the default .NET Core Linux containers run as root.

Reply:
It looks like the Users group have permissions to create a folder and write to subfolders. (At least that's the case on my system and documentation from what I can see so far). My concern is that the first user would get to create the settings without elevation, and then the subsequent users would either not have any rights to those settings or be able to change them freely.

Reply:
Discussed this offline a bit... seems that folders and files created under %ProgramData% are writable by admins AND the user that created them; all other users only have read-only and execute permissions to them.

Reply:
While I haven't found explicitly official documentation on this, [this blog post](https://docs.microsoft.com/en-us/archive/blogs/patricka/where-should-i-store-my-data-and-configuration-files-if-i-target-multiple-os-versions#per-machine-configuration-data) summaries where certain types of data should be stored. It says per-machine configuration data should be store under %SystemDrive%\ProgramData